src/database/DataTypes/TransactionItem.js

```diff
@@ -273,6 +273,43 @@ export class TransactionItem extends Realm.Object {
       });
     });
   };
+
+  /**
+   * Sets the doses for all underlying transaction batches. Rather than applying
+   * all doses in FEFO, apply an even distribution of doses over the transaction
+   * batches.
+   * @param {Number} value The number of doses to set for this item
+   */
+  setDoses(database, value) {
+    const dosesToSet = Math.min(value, this.totalQuantity * this.dosesPerVial);
+    const dosesToAssignToEachBatch = dosesToSet / this.totalQuantity;
+
+    this.resetAllDoses(database);
+    this.batches.sorted('expiryDate', false).forEach(batch => {
+      const { totalQuantity: thisBatchesQuantity } = batch;
+      batch.setDoses(database, Math.floor(dosesToAssignToEachBatch * thisBatchesQuantity));
+    });
+  }
```

Comment:
this behaviour is from the old vaccines branch. This is for issuing in a customer invoice, which is a single line for multiple batches, which for normal quantity is dispensed first expiry, first out. For doses, the behaviour was intended to divide the number of doses evenly over the number of batches for that line